storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -5116,15 +5115,15 @@ static bool xtrabackup_init_temp_log(void) {
     goto error;
   }
 
-  log_format = (Log_format)mach_read_from_4(log_buf + LOG_HEADER_FORMAT);
-  xtrabackup_original_log_format = log_format;
+  log_format =
+      static_cast<Log_format>(mach_read_from_4(log_buf + LOG_HEADER_FORMAT));
+  log_detected_format = log_format;
   if (log_format == Log_format::VERSION_8_0_28 ||
       log_format == Log_format::VERSION_8_0_19) {
     // We interpret v4 & v5 as v6 by writing proper start lsn header later in
     // this function
     log_format = Log_format::VERSION_8_0_30;
```

Comment:
How mortifying this in-memory variable will fix log_files_validate_format() to not throw error? There it will get format from file. isn't it?

Reply:
What fixes log_files_validate_format is the if below:
```
if (log_format == Log_format::VERSION_8_0_28 ||
      log_format == Log_format::VERSION_8_0_19) {
    // We interpret v4 & v5 as v6 by writing proper start lsn header later in
    // this function
    log_format = Log_format::VERSION_8_0_30;
    ```

Reply:
ah, we rewrite header then?